network/discovery/enode.go

```diff
@@ -33,38 +32,49 @@ func createLocalNode(privKey *ecdsa.PrivateKey, storagePath string, ipAddr net.I
 	return localNode, nil
 }
 
-// addAddresses adds configured address and/or dns if configured
+// addAddresses configures node addressing by prioritizing HostDNS over HostAddress.
+// Uses resolved DNS IP if available, otherwise falls back to HostAddress.
+// Returns error if address resolution fails or addresses are invalid.
 func addAddresses(localNode *enode.LocalNode, hostAddr, hostDNS string) error {
-	if len(hostAddr) > 0 {
-		hostIP := net.ParseIP(hostAddr)
-		if hostIP.To4() == nil && hostIP.To16() == nil {
-			return fmt.Errorf("invalid host address given: %v", hostAddr)
-		}
-		localNode.SetFallbackIP(hostIP)
-		localNode.SetStaticIP(hostIP)
-	}
+	// Try DNS first
 	if len(hostDNS) > 0 {
 		ips, err := net.LookupIP(hostDNS)
 		if err != nil {
-			return errors.Wrap(err, "could not resolve host address")
+			return fmt.Errorf("could not resolve host DNS: %s, %w", hostDNS, err)
 		}
+
 		if len(ips) > 0 {
 			firstIP := ips[0]
+			localNode.SetStaticIP(firstIP)
 			localNode.SetFallbackIP(firstIP)
+
+			return nil
 		}
 	}
+
+	// Use HostAddress as fallback
+	if len(hostAddr) > 0 {
```

Comment:
Should we handle the case when both `hostDNS` and `hostAddr` are empty?

Reply:
no need to update these conditions. The node will handle it automatically, its also tested here: https://github.com/ssvlabs/ssv/blob/0bc5eeff0daef49ae053eba54c6822b8b71a1d11/network/discovery/service_test.go#L370